src/spikeinterface/core/basesorting.py

```diff
@@ -6,8 +6,16 @@
 import numpy as np
 
 from .base import BaseExtractor, BaseSegment
+from .core_tools import spike_vector_to_dict
 from .waveform_tools import has_exceeding_spikes
 
+try:
+    import numba
+
+    HAVE_NUMBA = True
+except:
+    HAVE_NUMBA = False
+
```

Comment:
I don't know if @samuelgarcia is going to like all the attempts at importing numba....

Reply:
I just did exactly what samuel and I did in postprocessing/correlograms!
I never liked this, but trying to keep the code consistant (and you don't like my elegant but complicated code)

Reply:
I meant in core. Last time I asked why SI didn't just make numba a core dependency he said because he didn't want to add more imports to core to keep it light and fast. 

Nothing against the code just warning you before you work too hard on it. :)

Reply:
The bare try except do adds to the import time though. That is, in the current status numba will be imported when you import anything from spikeinterface which is undesirable.

At the moment, we either add a more complicated logic like this:
https://github.com/SpikeInterface/spikeinterface/blob/329197618a9b48ef876d1e8b8e79f07f4abf5e49/src/spikeinterface/comparison/comparisontools.py#L111-L232

Which is too complex.
or
We eat some import time.

My preference is for the former. 

But it would be good to come up with something simpler that avoids the import time with numba. For most stuff just moving the import inwards to the function solves the problem.

Reply:
I have to say, I personally hate having a function inside a function.
But if you prefer it that way, I can make it.

Reply:
No, the preferred way right now is to use the try-except block. I was experimenting above.

I am saying that the try-except block adds to the import time.